air-script-core/src/expression.rs

```diff
@@ -1,17 +1,11 @@
-use super::{Identifier, ListFolding, MatrixAccess, TraceAccess, TraceBindingAccess, VectorAccess};
+use super::{BindingAccess, Identifier, ListFolding, TraceAccess, TraceBindingAccess};
 
 /// Arithmetic expressions for evaluation of constraints.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub enum Expression {
     Const(u64),
-    /// Represents any named constant or variable.
-    Elem(Identifier),
-    /// Represents an element inside a constant or variable vector. [VectorAccess] contains the
-    /// name of the vector and the index of the element to access.
-    VectorAccess(VectorAccess),
-    /// Represents an element inside a constant or variable matrix. [MatrixAccess] contains the
-    /// name of the matrix and indices of the element to access.
-    MatrixAccess(MatrixAccess),
+    /// Represents a reference to all or part of a constant, variable, or trace binding.
+    BindingAccess(BindingAccess),
     TraceAccess(TraceAccess),
     TraceBindingAccess(TraceBindingAccess),
     /// Represents a random value provided by the verifier. The first inner value is the name of
```

Comment:
- I'm fine with renaming `BindingAccess` to `SymbolAccess`, but I think it would be cleaner to do it in a separate PR.
- I agree that I'd like to remove `Rand` and `TraceBindingAccess` and just use `BindingAccess` (or `SymbolAccess`), but I think this will require a lot of changes and should be addressed in a separate PR as well. Adding `offset` to the `BindingAccess` struct should resolve the complication I mentioned in the PR description.
- We shouldn't remove `TraceAccess`, as it serves a different function. It is the type used in Value nodes which are leaves in the `AlgebraicGraph`. However, we can remove it from use in the parser so it is only used in the IR. I would also do this in a separate PR.